praw/exceptions.py

```diff
@@ -1,37 +1,87 @@
 """PRAW exception classes.
 
-Includes two main exceptions: :class:`.APIException` for when something goes
-wrong on the server side, and :class:`.ClientException` when something goes
-wrong on the client side. Both of these classes extend :class:`.PRAWException`.
+Includes two main exceptions: :class:`.RedditAPIException` for when something
+goes wrong on the server side, and :class:`.ClientException` when something
+goes wrong on the client side. Both of these classes extend
+:class:`.PRAWException`.
 
 All other exceptions are subclassed from :class:`.ClientException`.
 
 """
-from typing import Optional
+from typing import List, Optional, TypeVar, Union
+
+_RedditErrorItem = TypeVar("_RedditErrorItem")
 
 
 class PRAWException(Exception):
     """The base PRAW Exception that all other exception classes extend."""
 
 
-class APIException(PRAWException):
-    """Indicate exception that involve responses from Reddit's API."""
+class RedditErrorItem:
+    """Represents a single error returned from Reddit's API."""
+
+    @property
+    def error_message(self):
+        """Get the completed error message string."""
+        error_str = "{}: '{}'".format(self.error_type, self.message)
+        if self.field:
+            error_str += " on field '{}'".format(self.field)
+        return error_str
 
-    def __init__(self, error_type: str, message: str, field: Optional[str]):
-        """Initialize an instance of APIException.
+    def __init__(
+        self, error_type: str, message: str, field: Optional[str] = None
+    ):
+        """Instantiate an error item.
 
         :param error_type: The error type set on Reddit's end.
         :param message: The associated message for the error.
         :param field: The input field associated with the error if available.
         """
-        error_str = "{}: '{}'".format(error_type, message)
-        if field:
-            error_str += " on field '{}'".format(field)
-
-        super().__init__(error_str)
         self.error_type = error_type
         self.message = message
-        self.field = field
+        self.field = "" if not bool(field) else field
+
+    def __eq__(self, other: Union[_RedditErrorItem, List[str]]):
+        """Check for equality."""
+        if isinstance(other, RedditErrorItem):
+            return (self.error_type, self.message, self.field) == (
+                other.error_type,
+                other.message,
+                other.field,
+            )
+        return False
+
+    def __repr__(self):
+        """Return repr(self)."""
+        return "{}(error_type={}, message={}, field={})".format(
```

Comment:
Wait, sorry, that’s wrong. Do this:

```suggestion
        return "{}(error_type={!r}, message={!r}, field={!r})".format(
```